buildlib/dockers/docker-compose-x86_64.yml

```diff
@@ -21,71 +21,99 @@ services:
-        OS_VERSION: 7
+        MOFED_VERSION: 5.0-1.0.0.0
+        MOFED_OS: rhel8.2
+        CUDA_VERSION: 11.4.0
+        OS_VERSION: 8
         ARCH: x86_64
-  centos8-mofed5-cuda11:
-    image: centos8-mofed5-cuda11:5
+  ubuntu18.04-mofed5-cuda11:
+    image: ubuntu18.04-mofed5-cuda11:3
     build:
       context: .
       network: host
-      dockerfile: centos-release.Dockerfile
+      dockerfile: ubuntu-release.Dockerfile
       args:
         MOFED_VERSION: 5.0-1.0.0.0
-        MOFED_OS: rhel8.2
+        UBUNTU_VERSION: 18.04
         CUDA_VERSION: 11.4.0
-        OS_VERSION: 8
+        NV_DRIVER_VERSION: 470
         ARCH: x86_64
-  ubuntu16.04-mofed5-cuda11:
-    image: ubuntu16.04-mofed5-cuda11:5
+  ubuntu20.04-mofed5-cuda11:
+    image: ubuntu20.04-mofed5-cuda11:3
     build:
       context: .
       network: host
       dockerfile: ubuntu-release.Dockerfile
       args:
         MOFED_VERSION: 5.0-1.0.0.0
-        UBUNTU_VERSION: 16.04
-        CUDA_VERSION: 11.2.0
+        UBUNTU_VERSION: 20.04
+        CUDA_VERSION: 11.4.0
+        NV_DRIVER_VERSION: 470
         ARCH: x86_64
-  ubuntu18.04-mofed5-cuda11:
-    image: ubuntu18.04-mofed5-cuda11:5
+  ubuntu22.04-mofed5-cuda11:
+    image: ubuntu22.04-mofed5-cuda11:3
+    build:
+      context: .
+      network: host
+      dockerfile: ubuntu-release.Dockerfile
+      args:
+        MOFED_VERSION: 5.4-3.6.8.1
+        UBUNTU_VERSION: 22.04
+        CUDA_VERSION: 11.7.0
+        NV_DRIVER_VERSION: 515
+        ARCH: x86_64
+  centos7-mofed5-cuda12:
+    image: centos7-mofed5-cuda12:3
+    build:
+      context: .
+      network: host
+      dockerfile: centos-release.Dockerfile
+      args:
+        MOFED_VERSION: 5.0-1.0.0.0
+        MOFED_OS: rhel7.6
+        CUDA_VERSION: 12.0.0
+        OS_VERSION: 7
+        ARCH: x86_64
+  ubuntu18.04-mofed5-cuda12:
+    image: ubuntu18.04-mofed5-cuda12:3
     build:
       context: .
       network: host
       dockerfile: ubuntu-release.Dockerfile
       args:
         MOFED_VERSION: 5.0-1.0.0.0
         UBUNTU_VERSION: 18.04
-        CUDA_VERSION: 11.4.0
+        CUDA_VERSION: 12.0.0
+        NV_DRIVER_VERSION: 525
         ARCH: x86_64
-  ubuntu20.04-mofed5-cuda11:
-    image: ubuntu20.04-mofed5-cuda11:5
+  ubuntu20.04-mofed5-cuda12:
+    image: ubuntu20.04-mofed5-cuda12:3
     build:
       context: .
       network: host
       dockerfile: ubuntu-release.Dockerfile
       args:
         MOFED_VERSION: 5.0-1.0.0.0
         UBUNTU_VERSION: 20.04
-        CUDA_VERSION: 11.4.0
+        CUDA_VERSION: 12.0.0
+        NV_DRIVER_VERSION: 525
         ARCH: x86_64
-  ubuntu22.04-mofed5-cuda11:
-    image: ubuntu22.04-mofed5-cuda11:5
+  ubuntu22.04-mofed5-cuda12:
+    image: ubuntu22.04-mofed5-cuda12:3
     build:
       context: .
       network: host
       dockerfile: ubuntu-release.Dockerfile
       args:
         MOFED_VERSION: 5.4-3.6.8.1
         UBUNTU_VERSION: 22.04
-        CUDA_VERSION: 11.7.0
+        CUDA_VERSION: 12.0.0
```

Comment:
What is about cuda 12.1? 
Pay attention there's newer NV_DRIVER_VERSION

Reply:
For X.Y versions of cuda, release is built on a minimal supported Y for X.

Reply:
NV_DRIVER_VERSION is set according to the cuda version, OS, and CPU architecture.
https://developer.nvidia.com/cuda-11-7-0-download-archive?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=22.04&target_type=runfile_local